src/server/ApiRoutes/Bib.js

```diff
@@ -79,16 +79,19 @@ export const fetchLocationUrls = codes => nyplApiClient()
   .then(client => client.get(`/locations?location_codes=${codes}`));
 
 const addLocationUrls = (bib) => {
-  const holdingCodes = bib.holdings ?
-    bib
-      .holdings
-      .map(holding => holding.location.reduce((acc, el) => acc.concat([el.code]), []))
-      .reduce((acc, el) => acc.concat(el), [])
-    : [];
+  const holdingCodes = [];
+
+  if (bib.holdings) {
+    bib.holdings.forEach(({ location }) => {
+      if (location) {
+        location.forEach(locationEntry => holdingCodes.push(locationEntry.code));
+      }
+    });
+  }
```

Comment:
My preference is to retain the declarative style (with `map` and `reduce` instead of `forEach`). If the problem is just that some holdings don't have the `location` property, we can add a default (so we would have `(holding.location || [])` instead of `holding.location`. I also think that the first reduce can be replaced with a `map`, which will make things simpler, so it would look something like this:

```
(bib.holdings || [])
  .map(holding => (holding.location || []).map(location => location.code))
  .reduce((acc, el) => acc.concat(el), [])
```

Reply:
I also prefer `map` and `reduce`, but I think the `forEach`s predate this bug fix by several months and I don't think bug fixes need to address existing style issues. (In fact, sometimes they shouldn't) Sounds like some light refactor we could separately ticket?